README.md
# custom-dark

CHANGELOG.md
CHANGELOG
=========

v0.43 stable (September 01, 2019)
---------------------------

Tested/ applied 'Master' over Mail-in-a-Box **v0.43 (September 01, 2019)** and no additional changes have been needed. Just works, too!

v0.42b stable (August 04, 2019)
---------------------------

Tested/ applied 'Master' over Mail-in-a-Box **v0.42b (August 03, 2019)** and no additional changes have been needed. Just works, too!

v0.42 stable (July 05, 2019)
---------------------------

Tested/ applied 'Master' over Mail-in-a-Box **v0.42 (July 4, 2019)** and no additional changes have been needed. Just works, too!

v0.41 beta (March 09, 2019)
---------------------------

## Custom-Dark theme:

* First release of **custom-dark** theme as a **Stand-alone** add-on for the latest official **Mail-in-a-Box** release (doesn't mean you can't try to apply it over a nearly earlier version, just test/ try it on a development/ test server first or right after you have created a server 'snapshot' to have available a point where to return if goes wrong).

**Setup:**

* To download and config. the **custom-dark** theme as **root**, run:

```
cd && git clone -b master https://github.com/just4t/custom-dark.git
cd custom-dark
/bin/bash ./config.sh && cd
```

* Then, to apply the **custom-dark** theme run:

```
/bin/bash /root/miab-themes/custom-dark/custom-dark.sh
```
* To apply/ restore the **default-light** theme run:

```
/bin/bash /root/miab-themes/custom-dark/default-light.sh
```

[**EDIT 190310-19:00 CET**] Following some client(s) request(s) decided to change the setup scripts to delete useless directories & files accordingly to the theme liked to have installed. Then, knowing all the directories & files created/ used to apply the desired theme will be deleted automatically by de setup itself, **If you wish to apply a different theme at anytime run setup again.**


**Screenshot:**

Watch a screen recording (gif): http://bit.ly/2SIiioL

<p align="center">
<img  align="center" width="95%" alt="Custom-Dark theme" src="https://5moqlg.dm.files.1drv.com/y4mKr_LGNHQUYWYNka8ez5swI2Vt6-JlZyLxAs7x9TG5F4s4XP9wKET3seNDCjLHXLWngK66J9SKIhHWYWSpLun6qjbBhus-y5lCb-pJ_FELrBoyfXIan9_yMETFjxpNkdl4IRD8N9wr0BRpXJ5KCNEadLhHKmG0F58K6kXLIsGzBXGCZKgz7PrIllaMywjhH92Iyn7a2MU1_8hLkibJzRxBA/001-custom-dark.png?psid=1">
</p>


config.sh
#!/bin/bash

# Cleaning / moving downloaded /miab-themes folder to /root/ path.
mv /root/custom-dark/miab-themes /root/

wait

sleep 5

# Making sh files executable.
chmod +x /root/miab-themes/custom-dark/custom-dark.sh
chmod +x /root/miab-themes/custom-dark/default-light.sh

wait

echo
echo =================================================
echo
echo Done! Post download config. settings have been applied as follows:
echo
echo - Moved /root/custom-dark/miad-themes folder to /root/
echo - Made sure scripts are CHMOD=+x [executables]
echo - Deleted the now useless /root/custom-dark folder
echo
echo To apply the CUSTOM-DARK theme run:
echo
echo sudo /bin/bash /root/miab-themes/custom-dark/custom-dark.sh
echo
echo Or to apply/ restore the DEFAULT-LIGHT theme run:
echo
echo sudo /bin/bash /root/miab-themes/custom-dark/default-light.sh
echo
echo =================================================
echo

# Cleanup after downloading theme from github and applied this config.
rm -rf /root/custom-dark


miab-themes/custom-dark/default-light.sh
#!/bin/bash

echo
echo
echo You have selected to restore the DEFAULT-LIGHT theme to your Mail-in-a-Box server ...
echo

sleep 2

#  Making sure we already have removed unneeded files & folders for this theme.
rm -f $/usr/local/lib/mailinabox/vendor/assets/bootstrap/css/custom.dark.*.css
rm -rf $/usr/local/lib/mailinabox/vendor/assets/images

wait

# 'status-checks' sed 'delete/append' patch to restore the default-light theme:
sed -i -e 's/#888/#733/1; s/#cc0/#770/1; s/#03ab22/#040/1' /root/mailinabox/management/templates/system-status.html

# 'index' sed 'delete/append' patch to restore the default-light theme:
sed -i -e 's/#888/#733/1; s/#cc0/#770/1; s/#03ab22/#040/1' /root/mailinabox/management/templates/system-status.html
sed -i '8d' /root/mailinabox/management/templates/index.html
sed -i '7a\\t\t<title>{{hostname}} - Mail-in-a-Box Control Panel</title>' /root/mailinabox/management/templates/index.html
sed -i '12d' /root/mailinabox/management/templates/index.html
sed -i '11a\\t\t<link rel="stylesheet" href="/admin/assets/bootstrap/css/bootstrap.min.css">' /root/mailinabox/management/templates/index.html
sed -i '14d' /root/mailinabox/management/templates/index.html
sed -i '13a\\t\tbody {' /root/mailinabox/management/templates/index.html
sed -i '165d' /root/mailinabox/management/templates/index.html
sed -i '164a\\t\t<p>This is a <a href="https://mailinabox.email">Mail-in-a-Box</a>.</p>' /root/mailinabox/management/templates/index.html

echo
echo Working ...

sleep 5

echo
echo =================================================
echo Your desired LIGHT theme is READY!... REBOOT your server NOW and visit your BOX admin panel through your browser as PRIVATE / INCOGNITO mode and/or refresh your browser by cleaning your currently booked cookies and browser cache, to see the final results inmediately. Enjoy It!
echo =================================================
echo

# deleting '/root/miab-themes'
rm -rf /root/miab-themes


miab-themes/custom-dark/custom-dark.sh
#!/bin/bash

echo
echo
echo You have selected to apply the CUSTOM-DARK theme to your Mail-in-a-Box server ...
echo

sleep 2

# Old custom-dark theme version & CSS cleanup & Making sure we have the directory 'images' where to save to
rm -f /usr/local/lib/mailinabox/vendor/assets/bootstrap/css/custom.dark.*.css
rm -rf /usr/local/lib/mailinabox/vendor/assets/images
mkdir -p /usr/local/lib/mailinabox/vendor/assets/images

# Move 'custom-dark' 'css' & 'images' files to their /usr/local/lib/mailinabox/vendor/assets path location
cp -f /root/miab-themes/custom-dark/css/custom.dark.*.css /usr/local/lib/mailinabox/vendor/assets/bootstrap/css/
cp -f /root/miab-themes/custom-dark/images/*.png /usr/local/lib/mailinabox/vendor/assets/images/

wait

# 'status-checks' sed 'delete/append' patch for custom-dark theme:
sed -i -e 's/#733/#888/1; s/#770/#cc0/1; s/#040/#03ab22/1' /root/mailinabox/management/templates/system-status.html

wait

# 'index' sed 'delete/append' patch for custom-dark theme:
sed -i '8d' /root/mailinabox/management/templates/index.html
sed -i '7a\\t\t<link rel="icon" href="/admin/assets/images/wicon-32.png" type="image/png"> <title>{{hostname}} - Control Panel</title>' /root/mailinabox/management/templates/index.html
sed -i '12d' /root/mailinabox/management/templates/index.html
sed -i '11a\\t\t<link rel="stylesheet" href="/admin/assets/bootstrap/css/bootstrap.min.css"> <link rel="stylesheet" href="/admin/assets/bootstrap/css/custom.dark.bootstrap.min.css"> <link rel="stylesheet" href="/admin/assets/bootstrap/css/custom.dark.bootstrap.css">' /root/mailinabox/management/templates/index.html
sed -i '14d' /root/mailinabox/management/templates/index.html
sed -i '13a\\t\tbody { color: #ffffff; background: #161616;' /root/mailinabox/management/templates/index.html
sed -i '165d' /root/mailinabox/management/templates/index.html
sed -i '164a\\t\t<p align="center"><a href="https://{{hostname}}/mail/" target="_blank"><img src="/admin/assets/images/wlogo-100.png" alt="wlogo-100" title="This is our Mail-in-a-Box powered webmail"/></a></p>' /root/mailinabox/management/templates/index.html

echo
echo Working ...

sleep 5

echo
echo =================================================
echo Your desired DARK theme is READY!... REBOOT your server NOW and visit your BOX admin panel through your browser as PRIVATE / INCOGNITO mode and/or refresh your browser by cleaning your currently booked cookies and browser cache, to see the final results inmediately. Enjoy It!
echo =================================================
echo

# deleting '/root/miab-themes'
rm -rf /root/miab-themes
